gradle-plugins/cordformation/src/main/kotlin/net/corda/plugins/Cordform.kt

```
@@ -0,0 +1,198 @@
+     */
+    @Suppress("unused")
+    @TaskAction
+    fun build() {
+        project.logger.info("Running Cordform task")
+        initializeConfiguration()
+        installRunScript()
+        nodes.forEach {
+            it.build()
+        }
+        generateAndInstallNodeInfos()
+    }
+
+    private fun initializeConfiguration() {
+        if (null != definitionClass) {
+            val cd = loadCordformDefinition()
+            cd.nodeConfigurers.forEach { nc ->
+                node {
+                    nc.accept(this)
+                    rootDir(directory)
+                }
+            }
+            cd.setup { nodeName -> project.projectDir.toPath().resolve(getNodeByName(nodeName)?.nodeDir?.toPath()) }
+        } else {
+            nodes.forEach {
+                it.rootDir(directory)
+            }
+        }
+    }
+
+    private fun fullNodePath(node: Node): Path = project.projectDir.toPath().resolve(node.nodeDir.toPath())
+
+    private fun generateAndInstallNodeInfos() {
+        generateNodeInfos()
+        installNodeInfos()
+    }
+
+    private fun generateNodeInfos() {
+        project.logger.info("Generating node infos")
+        val generateTimeout = 120L
+        val processes = nodes.map { node ->
+            project.logger.info("Generating node info for ${fullNodePath(node)}")
+            Pair(node, ProcessBuilder("java", "-jar", Node.nodeJarName, "--just-generate-node-info")
+                    .directory(fullNodePath(node).toFile())
+                    .redirectErrorStream(true)
+                    // InheritIO causes hangs on windows due the gradle buffer also not being flushed.
+                    // Must redirect to output or logger (node log is still written, this is just startup banner)
+                    .redirectOutput(File(project.buildDir, "tmp/node-log.txt"))
+                    .start())
+        }
+        try {
+            processes.forEach { (node, process) ->
```

This didn't exist before and would just hang or silently fail.